lib/internal/process/warning.js

```diff
@@ -91,6 +96,11 @@ function doEmitWarning(warning) {
 
 function onWarning(warning) {
   if (!(warning instanceof Error)) return;
+
+  const disableWarnings = getOptionValue('--disable-warnings');
+  if ((warning.code && ArrayPrototypeIncludes(disableWarnings, warning.code)) ||
```

Comment:
Can you check if `disableWarnings` exist as the first boolean expression in this check for early quitting the check?